snowshu/core/graph.py

```diff
@@ -298,49 +298,28 @@ def _process_downstream_relation_set(
                                **relationship['edge_attributes'])
         return graph
 
-    def get_graphs(self) -> tuple:
+    def get_connected_subgraphs(self) -> tuple:
         """ Generates the set of (weakly) connected components of the object's graph
 
             Returns:
                 tuple of connected subgraphs of the original processing graph
         """
-        # TODO this function is manually finding the connected components of the main graph
-        # Can likely be simplified if not entirely replaced by get_weakly_connected_components()
 
         if not isinstance(self.graph, networkx.Graph):
-            raise ValueError(
-                'Graph must be built before SnowShuGraph can get graphs from it.')
-        # get isolates first
-        isodags = list(networkx.isolates(self.graph))
-        logger.debug(f'created {len(isodags)} isolate dags.')
-        # assemble undirected graphs
-        ugraph = self.graph.to_undirected()
-        ugraph.remove_nodes_from(isodags)
-        node_collections = self._split_dag_for_parallel(ugraph)
-
-        dags = [networkx.DiGraph() for _ in range(len(isodags))]
-        for graph, node in zip(dags, isodags):
-            graph.add_node(node)
-
-        for collection in node_collections:
-            dags.append(networkx.subgraph(self.graph, collection))
+            raise ValueError('Graph must be built before SnowShuGraph can get graphs from it.')
+
+        graph = self.graph.to_directed()
```

Comment:
Is there a reason we need to specify `to_directed` here? The graph that was built should already be a digraph (that disallows multiple edges between two nodes)

Reply:
That makes sense. We should probably fix those tests and remove the graph modification in the function. At no point should we be creating an undirected graph currently.

The tests were likely created using an undirected graph since the original `get_graphs` function called `graph.to_undirected()`